contracts/rootchain/RootChain.sol

```diff
@@ -260,7 +269,10 @@ contract RootChain is Ownable {
         exit memory exit_ = txExits[priority];
         require(exit_.state == ExitState.Pending, "no pending exit to challenge");
 
-        // confirm challenging transcations inclusion and challenge the exiting transaction
+        // must be a direct spend
+        require(ensureMatchingInputs(exitingTxPos, txList), "challenging transaction does not spend the deposit or is a grand child transaction");
```

Comment:
Can you add tests to for this functionality?

Reply:
Its a simple require statements. There is nothing to test really. The existing tests already cover `ensureMatchingInputs`

Reply:
Do they test the failing case?

Reply:
I think it's clear when it would fail. Any transaction that doesn't have a matching input. It's like providing a test for when the length of the transaction is 17.

Reply:
Ok sure.